src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Emitter.Helpers.cs

```diff
@@ -0,0 +1,328 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Extensions.Configuration.SourceGeneration
+{
+    public sealed partial class ConfigurationBinderSourceGenerator
+    {
+        private sealed partial class Emitter
+        {
+            private const string GeneratorProjectName = "Microsoft.Extensions.Configuration.Binder.SourceGeneration";
+
+            private static DiagnosticDescriptor TypeNotSupported { get; } = new DiagnosticDescriptor(
+                id: "SYSLIB1091",
+                title: new LocalizableResourceString(nameof(SR.TypeNotSupportedTitle), SR.ResourceManager, typeof(FxResources.Microsoft.Extensions.Configuration.Binder.SourceGeneration.SR)),
+                messageFormat: new LocalizableResourceString(nameof(SR.TypeNotSupportedMessageFormat), SR.ResourceManager, typeof(FxResources.Microsoft.Extensions.Configuration.Binder.SourceGeneration.SR)),
+                category: GeneratorProjectName,
+                defaultSeverity: DiagnosticSeverity.Info,
+                isEnabledByDefault: true);
+
+            // Unlike sourcegen warnings, exception messages should not be localized so we keep them in source.
+            private static class ExceptionMessages
+            {
+                public const string TypeNotSupported = "Unable to bind to type '{0}': '{1}'";
+            };
+
+            private static bool TypeIsEnum(ITypeSymbol type) => type is INamedTypeSymbol { EnumUnderlyingType: INamedTypeSymbol { } };
```

Comment:
>private static bool TypeIsEnum(ITypeSymbol type) => type is INamedTypeSymbol { EnumUnderlyingType: INamedTypeSymbol { } }; [](http://example.com/codeflow?start=12&length=122)

I suggest move such method and similar method `UseStringBasedParse` to a helper class. It is possible we can share it across different source generators.